src/main/java/me/karakelley/http/handlers/StaticFilesHandler.java

```diff
@@ -61,5 +67,17 @@ private Response serveFilesResponse(Response response, String requestedResource)
     }
     return resources;
   }
+
+  private List<String> lookForIndexFile(String requestedResource) {
+    List<File> files = publicDirectory.getDirectoriesAndFiles(requestedResource);
+    return files.stream()
+            .map(File::getName)
+            .filter(fileName -> fileName.equals(ROOT_INDEX_FILE))
+            .collect(Collectors.toList());
+  }
+
+  private boolean indexFileIsPresent(List<String> files) {
+    return files.size() > 0;
```

Comment:
This is a sure sign you want `Optional` rather than a `List`. An optional is just a way to represent a value that is either there or not...

```
Optional.ofNullable(42); //=> Some(42)
Optional.ofNullable(null); //=> None
```

You can also think of an optional as a `List` that can only ever be empty or have one element.

Reply:
I know you don't have much experience with `Optional`, so I want to clarify this comment a bit. `ofNullable` is a constructor—it's a way of "wrapping" a value inside an `Optional`. A different example (pseudo coding a bit here between Java and Scala) could be a function like `parseInt` with a type signature `String -> Optional<Integer>`. It could operate like this:
```
x = "23"
y = "hello"
xAsInt = parseInt(x) //=> Some(23)
yAsInt = parseInt(y) //=> None
```
What's great about this is that anything calling `parseInt` is forced, by the compiler, to address the possibility of `parseInt` not having returned a valid integer. This is because the _types_ of `xAsInt`  and `yAsInt` are not `Integer`/`null`, but instead both are `Optional<Integer>` (with one "under the hood" being a `Some<Integer>` and the other being `None`).

What do I mean by "forced to address the possibility"? Well, operations like `+` and `-` are not defined on `Optional<Integer>`—instead, you need to "unwrap" them:
```
def maybeAddFive(z: Optional<Integer>): Integer = {
  z match {
    case Some(a) => a + 5
    case None => // throw an exception, use some default, whatever... but you have to choose something!
  }
}
```

Reply:
I don't understand how this can be used in place of what I have though. I'm checking to see if a file (index.html) was found, if so do this, otherwise do this instead.